### PR TITLE
Redesign Flanger

### PR DESCRIPTION
**Pull request by RebeccaDeField**
Related topics: #2831 #2978 

In this pull request I have:

- Updated the knob alignment
- Changed Invert text to be aligned left next to the LED
- Resized background so that the space around the elements is even

**Before**
fe9252d0-66cf-11e6-8614-ddea238f94b4

**After**
unnamed

@Umcaruje @BaraMGB @LocoMatt

### CODE REVIEW

File: plugins/Flanger/FlangerControlsDialog.cpp [hunk 1]
```diff
@@ -38,24 +38,24 @@ FlangerControlsDialog::FlangerControlsDialog( FlangerControls *controls ) :
...
 	lfoFreqKnob->setModel( &controls->m_lfoFrequencyModel );
 	lfoFreqKnob->setLabel( tr( "Lfo Hz" ) );
 	lfoFreqKnob->setHintText( tr ( "Lfo:" ) , "Hz" );
 
 	Knob * lfoAmtKnob = new Knob( knobBright_26, this );
-	lfoAmtKnob->move( 86,10 );
+	lfoAmtKnob->move( 84,10 );
 	lfoAmtKnob->setVolumeKnob( false );
 	lfoAmtKnob->setModel( &controls->m_lfoAmountModel );
 	lfoAmtKnob->setLabel( tr( "Amt" ) );
```
Comment (Umcaruje) [inline plugins/Flanger/FlangerControlsDialog.cpp]: I'd rename `Amt` to `Amnt`, as that's what its called in the Peak Controller and LFO controller. Furthermore, I believe we should convert the labels to all caps, to be consistent with the other plugins: 
screenshot from 2016-08-29 01 48 58

Comment (zonkmachine) [inline plugins/Flanger/FlangerControlsDialog.cpp]: I think Amt is a common abbreviation for Amount and Amnt isn't. Amnt is even not recognized by this editor. So, maybe change the other ones?

Comment (RebeccaDeField) [inline plugins/Flanger/FlangerControlsDialog.cpp]: @BaraMGB I believe you worked on the Peak Controller and LFO controller. Do you have an opinion about this?

Comment (tresf) [inline plugins/Flanger/FlangerControlsDialog.cpp]: `AMNT` is 4 letters; more consistent with the majority of the knobs to the likes of already abbreviated `DCAY`, `FREQ`, `GAIN` and as a bonus it is less code changes.  At the end of the day, we'll always be faced with translation problems with any abbreviated words which causes problems like #2599, but for now, let's keep it `AMNT`.

**Comment by BaraMGB** [comment]
I agree with @tresf

File: plugins/Flanger/FlangerControlsDialog.cpp (hunk 1 above)
Comment (BaraMGB) [inline plugins/Flanger/FlangerControlsDialog.cpp]: I agree with @tresf

Comment (jasp00) [inline plugins/Flanger/FlangerControlsDialog.cpp]: The abbreviation for amount does not have to be 4 letters necessarily. There are knobs with 3 (`VOL`, `PAN`) and with 5 (`RIGHT`, `RATIO`). In contrast with `DCAY`, decay has `DEC` too. Regarding amount, `AMT` is in the [Collins dictionary](http://www.collinsdictionary.com/dictionary/english/amt); where is `AMNT` used?

Comment (tresf) [inline plugins/Flanger/FlangerControlsDialog.cpp]: So as long as we have a button labeled `TRES` for threshold, I think `AMT` is the least of our worries but I agree, if we use `DEC` in other places, we should standardize that as well, perhaps through a centralized translation file.

Comment (jasp00) [inline plugins/Flanger/FlangerControlsDialog.cpp]: We should change `tr( "Amt" )` with `ABBR( Amount )` and keep abbreviations in a header:
```c++
#define ABBR_Amount "AMT"
#define ABBR( str ) tr( ABBR_ ## str )
```

Comment (RebeccaDeField) [inline plugins/Flanger/FlangerControlsDialog.cpp]: Thank you @zonkmachine  @tresf @jasp00 for the input. From reading over the comments here I'm not sure whether we have come to a final decision about keeping it as AMT or changing it to AMNT?

Comment (tresf) [inline plugins/Flanger/FlangerControlsDialog.cpp]: @RebeccaDeField please keep it `AMNT` for now to move this along and remain consistent with the rest of the software.  The task of standardizing our abbreviations (and thus changing `AMNT` to `AMT`) is out of scope and will be handled as a separate initiative.

Comment (RebeccaDeField) [inline plugins/Flanger/FlangerControlsDialog.cpp]: Sounds good. I will make that change along with the other suggestions from, @Umcaruje as soon as I can :+1:

**Comment by RebeccaDeField** [comment]
I noticed that my fork has some errors and while fixing them, I accidentally wiped the commit for this PR, which closed it automatically. I have re-committed the files with some of the changes and reopened this PR :sweat_smile:

**Comment by RebeccaDeField** [comment]
> The LED label should stay as is, to be consistent with the peak controller.

When you say "stay as is" could you clarify about which part you want to stay the same? Do you mean that the label, the position, etc?

@Umcaruje I know you're away right now so no pressure on responding to this.

**Comment by Umcaruje** [comment]
I meant that the LED labels should stay the same case, e.g. Sentence case,
as I suggested the other labels to go all caps to be consistent. The
alignment looks good to me.

Regards from Greece

**Comment by RebeccaDeField** [comment]
@Umcaruje Well I'm glad to hear that is what you meant, because that is what I did :smile: 

I believe I have now completed all of the requested changes to the Flanger:
unnamed

@tresf @Umcaruje @BaraMGB

**Comment by Umcaruje** [comment]
"LFO HZ" sounds weird to me. Its the frequency of the LFO, so maybe "FREQ"
could be more suitable.

Also what is the Regeneration? Is it another name for feedback? Anyways,
I'll test this out when I have pc access.

**Comment by zonkmachine** [comment]
> "LFO HZ" sounds weird to me. Its the frequency of the LFO, so maybe "FREQ" could be more suitable.

I thought about this too. The two most common labels I've seen is: SPEED and RATE. RATE is what Moog and Electro-Harmonix uses. If you want to be that hip...

**Comment by Umcaruje** [comment]
RATE sounds good to me.

**Comment by RebeccaDeField** [comment]
> RATE sounds good to me. :+1: 1  

I'm on it :+1:

**Comment by RebeccaDeField** [comment]
@Umcaruje This is ready for review when you are available.

File: plugins/Flanger/FlangerControlsDialog.cpp [hunk 2]
```diff
@@ -38,45 +38,45 @@ FlangerControlsDialog::FlangerControlsDialog( FlangerControls *controls ) :
...
 
 	TempoSyncKnob * lfoFreqKnob = new TempoSyncKnob( knobBright_26, this );
-	lfoFreqKnob->move( 53,10 );
+	lfoFreqKnob->move( 48,10 );
 	lfoFreqKnob->setVolumeKnob( false );
 	lfoFreqKnob->setModel( &controls->m_lfoFrequencyModel );
-	lfoFreqKnob->setLabel( tr( "Lfo Hz" ) );
-	lfoFreqKnob->setHintText( tr ( "Lfo:" ) , "Hz" );
+	lfoFreqKnob->setLabel( tr( "RATE" ) );
+	lfoFreqKnob->setHintText( tr ( "Rate:" ) , "" );
```
Comment (jasp00) [inline plugins/Flanger/FlangerControlsDialog.cpp]: I would not remove the unit: `tr ( "Rate:" ) , "Hz"`.

**Comment by RebeccaDeField** [comment]
@jasp00 Done :+1:

File: plugins/Flanger/FlangerControlsDialog.cpp (hunk 2 above)
Comment (RebeccaDeField) [inline plugins/Flanger/FlangerControlsDialog.cpp]: @jasp00 Done :+1:

**Comment by tresf** [comment]
@RebeccaDeField zero commits auto-closes a PR.  Please feel free to reopen a new one once your commit count is above 0.   No worries BTW, this is a github feature, so we just have to deal with it.

**Comment by RebeccaDeField** [comment]
@tresf Yes, unfortunately I had to reset because of a bad tutorial messing up my fork. I guess that is what you get from learning git from trial and error :P I will reopen once issues have been resolved.

**Comment by RebeccaDeField** [comment]
@tresf Issues resolved. PR reopened.

@Umcaruje Regen has been updated to FDBK as requested.

Mentioning some people that might have an opinion about the label change before it is merged @jasp00 @BaraMGB @zonkmachine.

File: plugins/Flanger/FlangerControlsDialog.cpp [hunk 3]
```diff
@@ -38,45 +38,45 @@ FlangerControlsDialog::FlangerControlsDialog( FlangerControls *controls ) :
...
+	lfoFreqKnob->setHintText( tr ( "Rate:" ) , "Hz" );
 
 	Knob * lfoAmtKnob = new Knob( knobBright_26, this );
-	lfoAmtKnob->move( 86,10 );
+	lfoAmtKnob->move( 85,10 );
 	lfoAmtKnob->setVolumeKnob( false );
 	lfoAmtKnob->setModel( &controls->m_lfoAmountModel );
-	lfoAmtKnob->setLabel( tr( "Amt" ) );
+	lfoAmtKnob->setLabel( tr( "AMNT" ) );
 	lfoAmtKnob->setHintText( tr ( "Amt:" ) , "" );
```
Comment (Umcaruje) [inline plugins/Flanger/FlangerControlsDialog.cpp]: This should be `Amount:` rather than `Amt:`

**Comment by RebeccaDeField** [comment]
@Umcaruje The hint text as well?

**Comment by Umcaruje** [comment]
@RebeccaDeField I was talking about the hint text, yeah, the knob label is good.

**Comment by BaraMGB** [comment]
I don't know. Do we really need such cryptical abbreviations?

**Comment by RebeccaDeField** [comment]
@Umcaruje Done.

@BaraMGB Thank you for the feedback. Would you prefer that it is switched back to Regen, that we abbreviate it differently or maybe have a different idea to propose?

**Comment by BaraMGB** [comment]
No, I think "feedback" is the right name for this knob in a flanger effect. But FDBK looks strange for me. But I fear I got no better idea at the moment. I'm just not a friend of abbreviations.

**Comment by tresf** [comment]
`FBACK`?

**Comment by RebeccaDeField** [comment]
I guess FB is another one, but that makes me think of Facebook ;)

**Comment by BaraMGB** [comment]
FEEDB, perhaps. I don't know...

**Comment by RebeccaDeField** [comment]
FBACK sounds like the best option to me. Tres likes it and bara don't seem to mind it. I will update the label.

**Comment by RebeccaDeField** [comment]
Ideas so far...

- FEEDB
- FBACK
- FDBK

_Reposting from Discord chat:_
It is also something to consider that FDBK is the only official abbreviation of feedback that I could find online. Maybe we can go forward with FDBK and then change it later if need be?

**Comment by RebeccaDeField** [comment]
_Reposting from Discord chat:_
@zonkmachine  FDBK works and I don't mind REGEN either.
@BaraMGB  For me it isn't so important. @Rebecca FDBK is good enough.
@Umcaruje  I like FDBK more haha

With that, I'm going to keep FDBK for now. If we want to make another PR to change it later, we can do that. If the flanger does not receive any additional changes in a couple days, I will merge. :+1:

**Comment by RebeccaDeField** [comment]
It's been two days without any further feedback or changes. Merged.

**Comment by RebeccaDeField** [comment]
It's been two days without any further feedback or changes. Merged.

**Comment by Umcaruje** [comment]
:+1:

**Pull request merged**